### PR TITLE
Enable OpenTelemetry logging in v6

### PR DESCRIPTION
**Pull request by 10xLaCroixDrinker**
Builds on #1084, pairs with #1059

## Description

Added otel transport

## Motivation and Context

Enable logs to be sent to OTel collector.

There is a [bug](https://github.com/open-telemetry/opentelemetry-js-contrib/issues/1275) that is currently blocking instrumentation, and a [limitation](https://github.com/Vunovati/pino-opentelemetry-transport/issues/20) on the otel transport that required a bit of a hack for sending otel logs to console in development. (Both documented in code comments)

## How Has This Been Tested?

Ran locally with prod sample sending logs to local collector. Yet to test in higher environments. Hold merging until then

- [ ] testing in higher environments complete

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [x] This change adds additional environment variable requirements for One App users.
- [x] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

Adds ability to use open telemetry for logs

### CODE REVIEW

File: src/server/utils/logging/config/otel.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import os from 'os';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+// import { registerInstrumentations } from '@opentelemetry/instrumentation';
+// import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+// import { FastifyInstrumentation } from '@opentelemetry/instrumentation-fastify';
+// import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
```
Comment (Matthew-Mallimo) [inline src/server/utils/logging/config/otel.js:22]:
```suggestion
```

File: src/server/utils/logging/config/otel.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// if (process.env.OTEL_EXPORTER_OTLP_LOGS_ENDPOINT) {
+//   const provider = new NodeTracerProvider();
+//   provider.register();
+//   registerInstrumentations({
+//     instrumentations: [
+//       new HttpInstrumentation(),
+//       new FastifyInstrumentation(),
+//     ],
+//   });
+// }
```
Comment (Matthew-Mallimo) [inline src/server/utils/logging/config/otel.js:45]:
```suggestion
```

File: src/server/utils/logging/logger.js [hunk 3]
```diff
@@ -15,20 +15,38 @@
  */
 
 import { argv } from 'yargs';
-import pino from 'pino';
```
Comment (Matthew-Mallimo) [inline src/server/utils/logging/logger.js:18]: is this exported both as the default and a named export?

**Review by Matthew-Mallimo — CHANGES_REQUESTED** [review]

File: src/server/utils/logging/logger.js (hunk 3 above)
Comment (10xLaCroixDrinker) [reply src/server/utils/logging/logger.js:18]: yes

File: src/server/utils/logging/config/otel.js (hunk 1 above)
Comment (JAdshead) [reply src/server/utils/logging/config/otel.js:22]: @10xLaCroixDrinker can these be removed ?

Comment (PixnBits) [reply src/server/utils/logging/config/otel.js:22]: https://github.com/fastify/fastify/pull/4470 is closed (merged) and it looks like https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1679#issuecomment-1723697660 is pretty close

Comment (10xLaCroixDrinker) [reply src/server/utils/logging/config/otel.js:22]: I'd rather keep all of this in for now

File: src/server/utils/logging/config/otel.js (hunk 2 above)
Comment (10xLaCroixDrinker) [reply src/server/utils/logging/config/otel.js:45]: I'd like to keep this in for now

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by JAdshead — APPROVED** [review]

**Pull request merged**